<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>SegwayRMP Demo</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1" colspan="2">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Commands</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <widget class="QPushButton" name="reset_integrators_button">
         <property name="text">
          <string>Reset Integrators</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="disable_motors_button">
         <property name="text">
          <string>Send Shutdown</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QPushButton" name="tractor_button">
         <property name="text">
          <string>Request Tractor</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QPushButton" name="balance_button">
         <property name="text">
          <string>Request Balance</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QPushButton" name="power_down_button">
         <property name="text">
          <string>Request Power Down</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QPushButton" name="balance_lockout_button">
         <property name="text">
          <string>Balance Lockout</string>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <widget class="QPushButton" name="balance_unlock_button">
         <property name="text">
          <string>Balance Unlock</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="Line" name="line_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="0" colspan="5">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="connect_button">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Connect</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QComboBox" name="connection_type">
        <item>
         <property name="text">
          <string>USB</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Serial</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="rmp_type">
        <property name="currentIndex">
         <number>2</number>
        </property>
        <item>
         <property name="text">
          <string>RMP 50</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>RMP 100</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>RMP 200</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>RMP 400</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="connection_id">
        <property name="minimumSize">
         <size>
          <width>300</width>
          <height>0</height>
         </size>
        </property>
        <property name="editable">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="refresh_button">
        <property name="text">
         <string>Refresh</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="3" colspan="2">
     <widget class="QGroupBox" name="groupBox_4">
      <property name="maximumSize">
       <size>
        <width>210</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Joystick</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QComboBox" name="joy_list_cb">
           <property name="maximumSize">
            <size>
             <width>180</width>
             <height>16777215</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QProgressBar" name="lv_pb">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>178</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>178</height>
            </size>
           </property>
           <property name="minimum">
            <number>-100</number>
           </property>
           <property name="value">
            <number>0</number>
           </property>
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Linear</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="lv_axis_cb">
               <property name="enabled">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QSlider" name="lv_scale_slider">
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="singleStep">
              <number>5</number>
             </property>
             <property name="pageStep">
              <number>25</number>
             </property>
             <property name="value">
              <number>25</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="tickPosition">
              <enum>QSlider::TicksBelow</enum>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>Angular</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="av_axis_cb">
               <property name="enabled">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QSlider" name="av_scale_slider">
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="singleStep">
              <number>5</number>
             </property>
             <property name="pageStep">
              <number>25</number>
             </property>
             <property name="value">
              <number>25</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="tickPosition">
              <enum>QSlider::TicksBelow</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="send_chb">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Send to Segway</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QProgressBar" name="av_pb">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="minimumSize">
          <size>
           <width>178</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>178</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="minimum">
          <number>-100</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="0" rowspan="2">
     <widget class="QGroupBox" name="groupBox">
      <property name="minimumSize">
       <size>
        <width>325</width>
        <height>515</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>275</width>
        <height>10000</height>
       </size>
      </property>
      <property name="title">
       <string>Segway Status</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="status_label">
         <property name="text">
          <string>Time Stamp: 
  Seconds: 0
  Nanoseconds: 0
Pitch: 0
Pitch Rate: 0
Roll: 0
Roll Rate: 0
Left Wheel Speed: 0
Right Wheel Speed: 0
Yaw Rate: 0
Servo Frames: 0
Integrated Left Wheel Position: 0
Integrated Right Wheel Position: 0
Integrated Forward Displacement: 0
Integrated Turn Position: 0
Left Motor Torque: 0
Right Motor Torque: 0
UI Battery Voltage: 0
Powerbase Battery Voltage: 0
Operational Mode: 0
Controller Gain Schedule: 0
Commanded Velocity: 0
Commanded Yaw Rate: 0
Motor Status: E-Stopped</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="1" colspan="4">
     <widget class="QGroupBox" name="groupBox_3">
      <property name="title">
       <string>Configuration</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0" colspan="3">
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Maximum Velocity (0.0 - 1.0)</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Maximum Acceleration (0.0 - 1 .0)</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label4">
           <property name="text">
            <string>Maximum Turn Rate (0.0 - 1.0)</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Limit Current (0.0 - 1.0)</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="current_limit_sb">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.050000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="max_vel_sb">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.050000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="max_accel_sb">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.050000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="max_turn_rate_sb">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.050000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0" colspan="2">
        <layout class="QFormLayout" name="formLayout_2">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::FieldsStayAtSizeHint</enum>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Gain Schedule</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="gain_schedule_cb">
           <item>
            <property name="text">
             <string>Light</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Tall</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Heavy</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="2">
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QPushButton" name="send_config_button">
           <property name="text">
            <string>Send Configurations</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
